### PR TITLE
Ability to transform rotate any feature

### PR DESCRIPTION
**Pull request by sivako**
Select transform mode -> Select any feature -> click once on map control -> move mouse near to desired point of the feature to use as pivot -> ctrl key hold and move mouse to rotate feature.

transform-rotate

### CODE REVIEW

**Comment by supersonicclay** [comment]
Nice! To keep nebula more stateless and to more closely match turf, how about this for the API:

```javascript
new EditableGeoJsonLayer({
  mode: 'transformRotate',
  modeConfig: {
    rotatePivot: [-123, 38], // lng, lat
  }
})
```

And if `modeConfig` isn't supplied, it defaults to centroid. Then a consuming application would be responsible for capturing and passing the pivot point.

**Comment by ibgreen** [comment]
Nit: Nested props defeat some of deck.gl's performance optimizations (e.g. they will always shallow compare `false`).

```
 modeConfig: {
    rotatePivot: [-123, 38], // lng, lat
  }
```

That said, I see several layer authors that prefer to use nested props, so maybe we'll need to figure something out to support it better.

**Comment by georgios-uber** [comment]
Basically what @supersonicclay is saying.

File: src/lib/layers/editable-geojson-layer.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import distance from '@turf/distance';
 import center from '@turf/center';
 import destination from '@turf/destination';
 import bearing from '@turf/bearing';
+import turftransformRotate from '@turf/transform-rotate';
```
Comment (georgios-uber) [inline src/lib/layers/editable-geojson-layer.js]: Nit: turftransformRotate => turfTransformRotate

**Comment by sivako** [comment]
Thank you @supersonicclay @ibgreen @georgios-uber for reviewing. 
I have added the modeConfig prop and removed the key event handlers.

File: package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "@turf/ellipse": ">=4.0.0",
     "@turf/helpers": ">=4.0.0",
     "@turf/point-to-line-distance": ">=4.0.0",
+    "@turf/transform-rotate": ">=4.0.0",
```
Comment (georgios-uber) [inline package.json]: Great

**Pull request merged**